### PR TITLE
Modified default lint options

### PR DESCRIPTION
**Pull request by deanshi**
- Added generating a baseline file if baseline file does not exist
- Added warningsAsErrors and abortOnError to be true
- Added checkDependencies as true

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,15 +22,15 @@ buildscript {
...
     jcenter()
     mavenLocal()
 }
 
 dependencies {
     compile gradleApi()
     compile localGroovy()
-
-    compile 'com.android.tools.build:gradle:2.3.0'
+    compile 'com.android.tools.build:gradle:3.0.1'
```
Comment (deanshi) [inline build.gradle:33]: A little bit concerned about this. I'm curious to see how many projects are still using 2.3.0.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Mon Apr 03 16:00:17 EDT 2017
+#Sat Mar 17 08:17:15 EDT 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (mpeng3) [inline gradle/wrapper/gradle-wrapper.properties]: Why not use gradle-4.3?

File: build.gradle (hunk 1 above)
Comment (mpeng3) [reply build.gradle:33]: I don't think it's a problem - if a project wants to use the new version of this library, they should upgrade their gradle version too.

**Review by mpeng3 — APPROVED** [review]
Looks good to me - any comments @f2rf2r ?

File: build.gradle (hunk 1 above)
Comment (f2rf2r) [reply build.gradle:33]: While we are at this, we should add a changelog

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (deanshi) [reply gradle/wrapper/gradle-wrapper.properties]: I think it auto-changed it to 4.1 when I updated. I can make that change to 4.3.

**Comment by deanshi** [comment]
Added a changelog, let me know if the formatting needs to change or if there is something I missed in each version.

I'll change the date of the current version once the final approval has gone through.

**Review by mpeng3 — APPROVED** [review]
LGTM! :shipit:

**Review by mpeng3 — APPROVED** [review]

**Pull request merged**